# AyushOhlyan

In [17]:
#importing liberaries

import pandas as pd  
import scipy.stats as st

# 1. Load The Data

In [18]:
df = pd.read_csv(r'C:\Users\ayush\Downloads\catalyst.csv')
df

Catalyst1  Catalyst2
0      91.50      89.19
1      94.18      90.95
2      92.18      90.46
3      95.39      93.21
4      91.79      97.19
5      89.07      97.04
6      94.72      91.07
7      89.21      92.75

# 2. Print head and tail

In [19]:
# print head and tail of the data
print("\nTop 5 columns\n\n",df.head())
print("\nLast 5 columns\n",df.tail())


Top 5 columns

    Catalyst1  Catalyst2
0      91.50      89.19
1      94.18      90.95
2      92.18      90.46
3      95.39      93.21
4      91.79      97.19

Last 5 columns
    Catalyst1  Catalyst2
3      95.39      93.21
4      91.79      97.19
5      89.07      97.04
6      94.72      91.07
7      89.21      92.75


# 2. Calculate the sample mean and sample standard deviation for both the          catalyst

In [20]:
mu_1, std_1 = df['Catalyst1'].mean(), df['Catalyst1'].std()
print('Mean and Standard deviation for 1st catalyst:-',mu_1,"&",std_1)
mu_2, std_2 = df['Catalyst2'].mean(), df['Catalyst2'].std()
print('Mean and Standard deviation for 2nd catalyst:-',mu_2,"&",std_2)

Mean and Standard deviation for 1st catalyst:- 92.25500000000001 & 2.385018718105646
Mean and Standard deviation for 2nd catalyst:- 92.73249999999999 & 2.983453176629871


# 3. The number of rows for each catalyst

In [21]:
n1, n2 = df['Catalyst1'].count(), df['Catalyst2'].count()
print('No. of rows for each catalyst:-',n1,",",n2)

No. of rows for each catalyst:- 8 , 8


# 4. Use the eight step hypothesis process:-

1.The parameter of interest are mu1 and mu2, the mean process yield using catalyst 1 and 2 respectively. We want to know if mu1-mu2=0. 

2.Null Hypothesis Ho: mu1-mu2=0 or Ho: mu1=mu2.

3.Alternative Hypothesis H(alpha): mu1!=mu2.

4.alpha=0.05.

5.The test statistic is: to=[bar(x1)-bar(x2)-0]/Sp*sqrt(1/n1 + 1/n2)

6.Reject Ho if to>t(0.025,14)=2.145 or if to<-t(0.025,14)=-2.145

In [22]:
st.t.ppf(1.95/2,14)

2.1447866879169273

7.Computation: we have bar(x1)=92.55,s1=2.39,n1=8,bar(x2)=92.733,s2=2.98,n2=8.

Then we find: s^(2)p=7.30 and sp=2.70 

With the help of fifth step we find 'to' which is equal to -0.35


8.Conclusions: Since -2.145<to = -0.35>2.145, the null hypothesis cannot be rejected. That is at the 0.05 significance level, we do not have strong evidence to conclude that catalyst 2 results in a mean yield that differs from the mean yield when catalyst 1 is used.

Alternatively, we can use scipy.stats, a submodule of scipy

In [23]:
st.ttest_ind(df['Catalyst1'],df['Catalyst2'])

Ttest_indResult(statistic=-0.3535908643461798, pvalue=0.7289136186068217)

The method returns the t statistic and p value.

# SUMMARY:-

:- First of all we load the data then, we print the head and tail of the data.

:- After, that we calculate the sample mean and sample standard deviation for both the catalyst.
   
    :- The mean and standard deviation for 1st catalyst is 92.25500000000001 and 2.385018718105646 respectively.
       
       :- The mean and standard deviation for 2nd catalyst is 92.73249999999999 and 2.983453176629871 respectively.
       
:- Then, we find out the number of rows for each catalyst which is (8,8)

:- Then, we use the eight hypothesis process:-
    
      :- In this we want to know mu1-mu2=0 or not, then we check null hypothesis then alternative hypothesis after that, we          compute and we find the conclusions.

# END OF THE PROJECT 

# THANK YOU